### DISCUSSION TITLE
All Legrand shutter switch proportional THE solution ;-)

### DISCUSSION THREAD

**Discussion by Pepeuch** [Koenkk/zigbee2mqtt#19048] [Devices]
hello, I updated my legran switches to the latest firmware dated No. 56. This now allows the switches to manage the partial opening/closing of the shutter after entering learning mode. when I use the H+C application it puts the switch in this mode, then asks me to keep it pressed until it closes completely, then come to the upper stop, letting it press up to the stop then release, the closed position until the first blade touches and we release and finally the upper stop a second time. then saves. I cannot find a similar function with Z2M or even any report of this function. Do you think that in the future there will be an evolution on this side? thank you in advance

**Comment by Pepeuch** [comment]
Sorry for this but I resolve the problem ( warning only availlable with firmware 56!!!) and found the solution and it works without the need for Z2M or ZHA other!!! the learning is mechanical and is recorded in the button directly.

Procedure:
-  put your shutter at the top stop wait for the button to click (30s)
- press stop until the light flashes orange (transition to learning mode)
- press down until the shutter closes completely and the soul is blocked 
- release the indicator light flashes let 
- press the up button until you reach the upper stop 
- release the indicator light flashes let 
- press down button until the first blade touches the bottom 
- release the indicator light flashes 
- press the up button until it reaches the top stop 
- release the indicator light flashes and turns green.


So work it with Z2M in proportionnal mode.

add this in this page: https://www.zigbee2mqtt.io/devices/067776.html

Thanks

**Reply by xontik** [reply]
Two years searching for another way to have proportional opening... with delays, end switch, distance sensors.

You fixed that in 10sec !
Thank you

**Comment by adnev** [comment]
Hello Pepeuch,
Thanks for your advice to enable the proportional position of the shutter. 
Could you indicates which version of the shutter switch are you using? There is the new  version with an A letter at the end (for example 067726A)
On the Legrand website it also indicates a step by step procedure to calibrates the shutter switch : https://assets.legrand.com/pim/NP-FT-GT/LE10318AD.pdf

In other post, @cyrilphoenix71 mentionned lot of difficulties to associate these new versions of the shutter switch (post : https://github.com/Koenkk/zigbee2mqtt/issues/16178)

**Comment by Pepeuch** [comment]
My version is 067776

**Reply by Nusephived** [reply]
Hello,

What's the difference between version 067726 and 067726A?
Why do you specify that it's only available with firmware 56? Did they discontinue the functionality after 56?

Thx for the guide btw :)

**Reply by Pepeuch** [reply]
hello, I expressed myself badly, and since then I have had some feedback of good functioning of the learning since version 38 and after, at the beginning we thought that it was only functional from a certain version.
